### PR TITLE
Migrate to monorepo

### PR DESCRIPTION
**Pull request by rekmarks**
In service of #104, this PR migrates this repo to a monorepo structure. The goal with this PR is to support splitting the existing config package into multiple packages. CI scripts have been altered to support the existence of multiple packages, but are otherwise unchanged.

After this is merged, numerous steps remain, including splitting the various configs into their own packages, migrating to GitHub Actions, and setting up automatic releases.

### CODE REVIEW

File: .circleci/scripts/deps-install.sh [hunk 1]
```diff
@@ -5,4 +5,10 @@ set -e
 set -u
 set -o pipefail
 
-yarn --frozen-lockfile --ignore-scripts --har
+cd ./packages
+
+# ref: https://github.com/koalaman/shellcheck/wiki/SC2044
+find . -mindepth 1 -maxdepth 1 -type d -exec sh -c '
+    cd "$1"
+    yarn --frozen-lockfile --ignore-scripts --har
```
Comment (Gudahtt) [inline .circleci/scripts/deps-install.sh:13]: Ideally we'd install dependencies _once_ at the top-level. That would eliminate the need for these two loops.

File: LICENSE [hunk 2]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (Gudahtt) [inline LICENSE:1]: Was this intentional? :thinking:

File: packages/base/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@metamask/eslint-config",
```
Comment (Gudahtt) [inline packages/base/package.json:2]: The manifest for the other packages seems to be missing :thinking:

Comment (Gudahtt) [reply packages/base/package.json:2]: Ah nevermind, I misunderstood. This is still a single-package monorepo.

File: .circleci/scripts/deps-install.sh (hunk 1 above)
Comment (rekmarks) [reply .circleci/scripts/deps-install.sh]: Hehe, yeah, it was only installing once even with multiple packages in my follow-up PR 😅 

f05e39f

File: lerna.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  "command": {
+    "publish": {
+      "registry": "https://registry.npmjs.org"
+    }
+  },
+  "packages": [
+    "packages/*"
+  ],
+  "npmClient": "yarn",
+  "useWorkspaces": true
```
Comment (Gudahtt) [inline lerna.json:12]: There's a pretty serious bug in Yarn v1 relating to workspaces: the `--frozen-lockfile` flag doesn't work properly. See here for details: https://github.com/yarnpkg/yarn/issues/4098

If we're going to use Yarn workspaces, we should update the `deps-install` script to omit the `--frozen-lockfile` flag and check manually whether there have been any lockfile changes (e.g. check to ensure the working tree is still unchanged).

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,110 @@
...
+.vuepress/dist
+
+# Serverless directories
+.serverless/
+
+# FuseBox cache
+.fusebox/
+
+# DynamoDB Local files
+.dynamodb/
```
Comment (Gudahtt) [inline .gitignore:107]: This seems like _a lot_ :thinking: What is all this stuff? I don't think we need to add gitignore entries for things that are far removed from the tools we're using.

Comment (rekmarks) [reply .gitignore:107]: It's just GitHub's default for Node 😛 

I'll go through it and git rid of some cruft.

Comment (Gudahtt) [reply .gitignore:107]: Ah I see - I hadn't realized.

The template repo has a decent one I think? Might be a good starting place.

Comment (rekmarks) [reply .gitignore]: 29639f2

File: lerna.json (hunk 4 above)
Comment (rekmarks) [reply lerna.json:12]: 0214dc0

File: .gitignore (hunk 5 above)
Comment (rekmarks) [reply .gitignore]: Strike that: 400e4e8

File: package.json [hunk 6]
```diff
@@ -1,40 +1,7 @@
...
-  "peerDependencies": {
-    "eslint": "^7.7.0",
-    "eslint-plugin-import": "^2.22.0"
-  }
+  "name": "root",
+  "private": true,
+  "workspaces": [
+    "packages/base"
+  ]
 }
```
Comment (Gudahtt) [inline package.json:7]: Should `lerna` be here as a dev dependency? :thinking:

File: yarn.lock [hunk 7]
```diff
@@ -15,9 +15,9 @@
   integrity sha512-np/lG3uARFybkoHokJUmf1QfEvRVCPbmQeUQpKow5cQ3xWrV9i3rUHodKDJPQfTVX61qKi+UdYk8kik84n7XOw==
 
 "@babel/highlight@^7.10.4":
-  version "7.13.8"
```
Comment (Gudahtt) [inline yarn.lock:18]: Were these changes accidental? I don't see any dependency changes in this PR

File: package.json (hunk 6 above)
Comment (rekmarks) [reply package.json:7]: Hm, I think `lerna` recommends that you install it globally. I can either add it as a dev dep or add a note in the readme?

File: yarn.lock (hunk 7 above)
Comment (rekmarks) [reply yarn.lock]: Yeah, that was accidental. I think this got rid of lockfile changes: f8ff174

File: package.json (hunk 6 above)
Comment (Gudahtt) [reply package.json:7]: Apparently not. From their README:

>Let's start by installing Lerna as a dev dependency of your project with npm.

Comment (rekmarks) [reply package.json:10]: 😅 56c43bc

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**